### DISCUSSION TITLE
Thymeleaf performance comparison

### DISCUSSION THREAD

**Discussion by ari** [thymeleaf/thymeleaf#836] [General]
I've come to this project from this link: https://github.com/mbosecke/template-benchmark

Although Thyme performs poorly there in comparison, it is not clear whether this is just old data and Thymeleaf is now faster, or whether poor configuration choices were made in those tests, or whether performance isn't a priority on this project and various caching layers considered a reasonable solution.

Someone submitted some improved updated data: https://github.com/mbosecke/template-benchmark/pull/15/files

While performance isn't always the most important goal, as long as it is fast enough for real use, is there any feedback about these results?

**Comment by goodale** [comment]
Looks like that analysis is based on a old version of thymeleaf   Someone has requested a merge request that would allow running in on the latest version but the analysis project looks stale or dead and the merge hasn’t been accepted

 

https://github.com/mbosecke/template-benchmark/pull/15 

 

From: Ari Maniatis ***@***.***> 
Sent: Tuesday, June 8, 2021 9:51 PM
To: thymeleaf/thymeleaf ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [thymeleaf/thymeleaf] Thymeleaf performance comparison (#836)

 

I've come to this project from this link: https://github.com/mbosecke/template-benchmark

Although Thyme performs poorly there in comparison, it is not clear whether this is just old data and Thymeleaf is now faster, or whether poor configuration choices were made in those tests, or whether performance isn't a priority on this project and various caching layers considered a reasonable solution.

Someone submitted some improved updated data: https://github.com/mbosecke/template-benchmark/pull/15/files

While performance isn't always the most important goal, as long as it is fast enough for real use, is there any feedback about these results?

**Comment by casid** [comment]
I've created a fork of this benchmark and upgraded all libraries to their latest version (Thymeleaf currently at 3.0.15.RELEASE). Thymeleaf got definitely faster! I also did a benchmark with more than one Thread and here Thymeleaf catches up a lot to the other engines, which is a more realistic benchmark for server-side template rendering.

You can have a look at the charts [here](https://github.com/casid/jte#performance).
